file_system.go

```diff
@@ -321,15 +321,27 @@ func (f *ftpFile) Sys() interface{} {
 	return f.raw
 }
 
-var lsRegex = regexp.MustCompile(`^\s*(\S)(\S{3})(\S{3})(\S{3})(?:\s+\S+){3}\s+(\d+)\s+(\w+\s+\d+)\s+([\d:]+)\s+(.+)$`)
+var (
+	lsRegex    = regexp.MustCompile(`^\s*(\S)(\S{3})(\S{3})(\S{3})(?:\s+\S+){3}\s+(\d+)\s+(\w+\s+\d+)\s+([\d:]+)\s+(.+)$`)
+	winLsRegex = regexp.MustCompile(`^\s*(\d{2}-\d{2}-\d{2}\s*\d{2}:\d{2}(?:PM|AM))\s*(<DIR>|\d+)\s*(\S+)$`)
+)
 
-// total 404456
-// drwxr-xr-x   8 goftp    20            272 Jul 28 05:03 git-ignored
 func parseLIST(entry string, loc *time.Location, skipSelfParent bool) (os.FileInfo, error) {
 	if strings.HasPrefix(entry, "total ") {
 		return nil, nil
 	}
 
+	if lsRegex.MatchString(entry) {
+		return parseLinuxLIST(entry, loc, skipSelfParent)
+	} else if winLsRegex.MatchString(entry) {
+		return parseWinLIST(entry, loc)
+	}
+	return nil, fmt.Errorf("unexpected LIST format %v", entry)
```

Comment:
```suggestion
	return nil, ftpError{err: fmt.Errorf(`unexpected LIST format: %s`, entry)}
```

To match the other error results here.